frontend/src/components/ShootDns/GDnsProviderRow.vue

```diff
@@ -176,58 +164,85 @@ export default {
     ]),
-    dnsProvider () {
-      return this.dnsProviders[this.dnsProviderId]
-    },
-    primary () {
-      return this.dnsPrimaryProviderId === this.dnsProviderId
-    },
     dnsSecrets () {
       return this.dnsSecretsByProviderKind(this.dnsProviderType)
     },
-    typeHint () {
-      return this.primary && !this.isNewCluster
-        ? 'Primary Provider type cannot be changed after cluster creation'
-        : ''
-    },
-    primaryReadonly () {
-      return !this.isNewCluster && this.primary
-    },
     dnsProviderType: {
       get () {
-        return this.dnsProvider.type
+        return this.dnsProvider?.type
       },
       set (value) {
         this.dnsProvider.type = value
         const dnsSecrets = this.dnsSecretsByProviderKind(value)
         const defaultDnsSecret = head(dnsSecrets)
-        this.dnsProvider.secretName = get(defaultDnsSecret, 'metadata.secretRef.name', null)
+        this.extensionDnsProviderSecret = defaultDnsSecret
       },
     },
-    dnsProviderSecret: {
+    extensionDnsProviderSecret: {
       get () {
-        return find(this.dnsSecrets, ['metadata.secretRef.name', this.dnsProvider.secretName])
+        const resourceName = this.dnsProvider.secretName
+        const secretName = this.getResourceRefName(resourceName)
+        return find(this.dnsSecrets, ['metadata.secretRef.name', secretName])
       },
       set (value) {
-        this.dnsProvider.secretName = get(value, 'metadata.secretRef.name', null)
+        this.deleteResource(this.dnsProvider.secretName)
+        const secretName = get(value, 'metadata.secretRef.name')
+        const resourceName = this.getDnsServiceExtensionResourceName(secretName)
+        this.dnsProvider.secretName = resourceName
+        this.setResource({
+          name: resourceName,
+          resourceRef: {
+            apiVersion: 'v1',
+            kind: 'Secret',
+            name: secretName,
+          },
+        })
       },
     },
-  },
-  watch: {
-    'v$.$invalid' (value) {
-      const oldValue = !this.dnsProvider?.valid
-      if (oldValue !== value && !this.dnsProvider.readonly) {
-        this.dnsProvider.valid = !value
-      }
+    excludeDomains: {
+      get () {
+        return get(this.dnsProvider, 'domains.exclude')
+      },
+      set (value) {
+        set(this.dnsProvider, 'domains.exclude', value)
+      },
+    },
+    includeDomains: {
+      get () {
+        return get(this.dnsProvider, 'domains.include')
+      },
+      set (value) {
+        set(this.dnsProvider, 'domains.include', value)
+      },
+    },
+    excludeZones: {
+      get () {
+        return get(this.dnsProvider, 'zones.exclude')
+      },
+      set (value) {
+        set(this.dnsProvider, 'zones.exclude', value)
+      },
+    },
+    includeZones: {
+      get () {
+        return get(this.dnsProvider, 'zones.include')
+      },
+      set (value) {
+        set(this.dnsProvider, 'zones.include', value)
+      },
+    },
+    filterSecretNames () {
```

Comment:
Same as above. This is property not a function. Do not use verbs for properties!

Reply:
changed to allowedSecretNames